Mathlib/Data/Nat/Init.lean

```diff
@@ -176,8 +148,9 @@ lemma forall_lt_succ : (∀ m < n + 1, p m) ↔ (∀ m < n, p m) ∧ p n := by
   simp only [Nat.lt_succ_iff, Nat.le_iff_lt_or_eq, or_comm, forall_eq_or_imp, and_comm]
 
 lemma exists_lt_succ : (∃ m < n + 1, p m) ↔ (∃ m < n, p m) ∨ p n := by
-  rw [← not_iff_not]
-  push_neg
+  classical
+  rw [← Decidable.not_iff_not]
+  simp [not_exists, not_or]
```

Comment:
Squeeze or `simpa` or add `forall_lt_succ` to the `simp` arguments or something

Reply:
Ah right, sorry for the nonterminal `simp`!